engine.js

```diff
@@ -7,8 +7,30 @@ var hmdajson = require('./lib/hmdajson'),
     _ = require('underscore'),
     brijSpec = require('brij-spec'),
     stream = require('stream'),
-    request = require('sync-request'),
-    moment = require('moment');
+    request = require('then-request'),
+    moment = require('moment'),
+    Q = require('q');
+
+var randomResultName = function() {
+    var text = '';
+    var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+
+    for (var i=0; i < 20; i++) {
```

Comment:
20 IS magic. It gave me enough random characters to ensure I would get something unique no matter how bad the implementation of Math.random();